src/Microsoft.Sbom.Extensions.DependencyInjection/RemapComponentDetectionErrorsToWarningsLogger.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Sbom;
+
+using System;
+using Serilog;
+using Serilog.Events;
+
+/// <summary>
+/// A class to remap Errors logged from ComponentDetection assemblies to Warnings.
+/// </summary>
+public class RemapComponentDetectionErrorsToWarningsLogger : ILogger
+{
+    private readonly ILogger logger;
+    private readonly Func<string?> stackTraceProvider;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RemapComponentDetectionErrorsToWarningsLogger"/> class.
+    /// Production constructor.
+    /// </summary>
+    public RemapComponentDetectionErrorsToWarningsLogger(ILogger logger)
+        : this(logger, () => Environment.StackTrace)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RemapComponentDetectionErrorsToWarningsLogger"/> class.
+    /// Testable constructor.
+    /// </summary>
+    /// <exception cref="ArgumentNullException"></exception>
+    public RemapComponentDetectionErrorsToWarningsLogger(ILogger logger, Func<string?> stackTraceProvider)
```

Comment:
nit: can we make the testable constructor internal? This also doesn't appear to be used right now, can we either add tests that exercise it or remove it?

Reply:
Unit tests have been added and the test constructor has been made internal